packages/neuron-ui/src/components/Receive/VerifyHardwareAddress.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useCallback, useRef, useState, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import Button from 'widgets/Button'
+import { ControllerResponse } from 'services/remote/remoteApiWrapper'
+import {
+  connectDevice,
+  getDevices,
+  getDeviceCkbAppVersion,
+  getDevicePublicKey,
+  DeviceInfo,
+  updateWallet,
+  getPlatform,
+} from 'services/remote'
+import { ErrorCode, clsx, errorFormatter, isSuccessResponse, useDidMount } from 'utils'
+import { CkbAppNotFoundException, DeviceNotFoundException } from 'exceptions'
+import { AddressPrefix, addressToScript, scriptToAddress } from '@nervosnetwork/ckb-sdk-utils'
+import Alert from 'widgets/Alert'
+import styles from './receive.module.scss'
+
+export interface VerifyHardwareAddressProps {
+  address: string
+  wallet: State.WalletIdentity
+}
+
+const toLongAddr = (addr: string) => {
+  try {
+    const script = addressToScript(addr)
+    const isMainnet = addr.startsWith(AddressPrefix.Mainnet)
+    return scriptToAddress(script, isMainnet)
+  } catch {
+    return ''
+  }
+}
+
+const verifyAddressEqual = (address: string, compared?: string) => {
```

Comment:
Does `compared` mean `the one to compare`, if so, naming it `compared` is a bit confusing, maybe `source` and `target` would be better